src/sync/atomic.rs

```diff
@@ -0,0 +1,39 @@
+// `AtomicU64` can only be used on targets with `target_has_atomic` is 64 or greater.
+// Once `cfg_target_has_atomic` feature is stable, we can replace it with
+// `#[cfg(target_has_atomic = "64")]`.
+// Refs: https://github.com/rust-lang/rust/tree/master/src/librustc_target
+cfg_if::cfg_if! {
+    if #[cfg(not(any(target_arch = "arm", target_arch = "mips", target_arch = "powerpc")))] {
```

Comment:
Many ` target_arch = "arm"` targets are `target_has_atomic = "64"`, but only some targets seems different. 
* [arm-linux-androideabi](https://github.com/rust-lang/rust/blob/master/src/librustc_target/spec/arm_linux_androideabi.rs#L7)
* [armv5te-unknown-linux-musleabi](https://github.com/rust-lang/rust/blob/master/src/librustc_target/spec/armv5te_unknown_linux_musleabi.rs#L23)

Reply:
cc #48 Should we support these targets?

Reply:
I would definitely support `arm-linux-androideabi`, I'm hard pressed to find a deployment target with `armv5` that is relevant for async-std.